blog/2024-02-09-private-api-gateway-aws.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Building a Private API Gateway in AWS with Winglang
+description: A step by step guid to build a private API Gateway in an AWS VPC with Winglang
+authors:
+  - skorfmann
+  - hasanaburayyan
+tags: [aws, api-gateway, vpc]
+hide_table_of_contents: false
+---
+
+This tutorial is a guide for creating a [secure](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-private-apis.html) [API Gateway](https://aws.amazon.com/api-gateway/) inside an AWS Virtual Private Cloud (VPC) using Winglang. A private API Gateway in AWS is used when you need to expose APIs that are only accessible within your Virtual Private Cloud (VPC) or through specific endpoints that you define. This setup is essential for internal or private applications where security and controlled access are crucial.
+
+Wing currently supports two programming languages: TypeScript and [Winglang](https://winglang.io). For the purposes of this tutorial, we'll use Winglang.
+
+Let's start by installing the Wing CLI (you'll need Node.js >= 18.x installed):
+
+```bash
+npm i -g winglang
+```
+
+You can check the CLI version like this:
+
+```bash
+wing --version
+0.57.22
+```
+
+> As you can see, Wing is still in pre-release, so expect some hiccups and don't be shy reporting issues or get help from the awesome people hanging out on the [Wing slack](https://t.winglang.io/slack).
+>
+
+Ok, now that we have the Wing CLI installed, let's create a new project using the `private-api` quickstart:
+
+```bash
+$ mkdir my-api
+$ cd my-api
+$ wing new private-api
+```
+
+---
+
+Let's check out what we now have in our project directory:
+
+```bash
+main.w
+wing.toml
+package-lock.json
+package.json
+```
+
+If we look at the `main.w` file, we'll see:
+
+```jsx
+bring cloud;
+bring http;
+
+let api = new cloud.Api();
```

Comment:
I kind of feel like the initial example should include a bit more elaborate. Maybe the API will send a request to a queue or something?

Reply:
We decided that we will leave the story as is and focus on the rest of the feedback. We will learn for the next use cases.